### DISCUSSION TITLE
Feature: Print an exception without Stack Trace

### DISCUSSION THREAD

**Discussion by gerardog** [spectreconsole/spectre.console#1485] [Ideas]
Sometimes in my console apps I use a convention, that defines that the app can raise a "business/user" exception that is meant to be displayed to the end user. 
This kind of error is so descriptive that a Stack Trace actually adds more noise and confusion instead of helping the end user.
In those scenarios I want to print the Exception type and description, but without a stack trace.

For example:

``` csharp
        static async Task<int> Main(string[] args)
        {
            try
            {
                return await DoSomething();
            }
            catch (ApplicationException ex)
            {
                AnsiConsole.WriteException(ex, ExceptionFormats.NoStackTrace); //  <-- New ExceptionFormat
                return 1;
            }
            catch (Exception ex)
            {
                // Show full stack trace
                AnsiConsole.WriteException(ex, ExceptionFormats.ShortenPaths | ExceptionFormats.ShortenTypes);
                return 1;
            }
        }
```

How does it sound? 

TBH... I have a [PR ready on my fork](https://github.com/gerardog/spectre.console/pull/1), all checks have passed.

**Comment by FrankRay78** [comment]
Given exceptions are already being formatted, and this is an extension of that, it seems like a reasonable proposal. Plus the code on your branch is clean and covered by tests.

I'm interested in what other people think about this idea too.

**Reply by patriksvensson** [reply]
I'm ok with adding this feature.

**Reply by gerardog** [reply]
😊 Great! I just created PR #1489

**Comment by patriksvensson** [comment]
Since this has been merged, I'll go ahead and close this as resolved.
